packages/plugin-types/src/deep_partial.ts

```diff
@@ -5,3 +5,14 @@
 export type DeepPartial<T> = {
   [P in keyof T]?: DeepPartial<T[P]>;
 };
+
+/**
+ * Makes all the properties of the entire nested object partial for Amplify generated configs
+ * instead of just the top level properties. Other properties are not changed.
+ */
+export type DeepPartialAmplifyGeneratedConfigs<T> = {
+  [P in keyof T]?: P extends 'analytics' | 'geo' | 'notifications'
+    ? T[P]
+    :
+    (T[P] extends object ? DeepPartialAmplifyGeneratedConfigs<T[P]> : Partial<T[P]>);
```

Comment:
I am trying to figure this out but prettier removes the parenthesis which are needed for the type to work as intended.

Doing this does not work:
```ts
export type DeepPartialAmplifyGeneratedConfigs<T> = {
  [P in keyof T]?: P extends 'analytics' | 'geo' | 'notifications'
    ? T[P]
    :
    // prettier-ignore
    (T[P] extends object ? DeepPartialAmplifyGeneratedConfigs<T[P]> : Partial<T[P]>);
};
```

Should we specifically ignore this file?

Reply:
It's better to just turn prettier off on one line with comments rather than ignoring the whole file.

Reply:
I was trying to do that like the example above but this does not work and prettier still removes the parenthesis. I'll keep trying.

Reply:
We can ignore this... I think I forgot to run build after making changes to the new type which led me to think the type doesn't work correctly without the parenthesis.